vllm/transformers_utils/tokenizer.py

```diff
@@ -129,7 +130,7 @@ def detokenize_incrementally(
     # The prefix text is necessary only to defeat cleanup algorithms in
     # the decode which decide to add a space or not depending on the
     # surrounding ids.
-    if not getattr(tokenizer, "added_tokens_encoder", {}):
+    if tokenizer.is_fast or not tokenizer.get_added_vocab():
```

Comment:
It seems `tokenizer.get_added_vocab` is executed many times. Maybe we can further optimize this by caching the added vocab.

Reply:
get added vocab is already cached for slow tokenizers (and fast tokenizers won't reach this condition)

Reply:
Oh got it. That's great. Thanks for letting me know!